staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4981,6 +4981,18 @@ type ServiceProxyOptions struct {
 }
 
 // ObjectReference contains enough information to let you inspect or modify the referred object.
+// New uses of this type are discouraged because of difficulty describing its usage when embedded in APIs.
+// Instead of using this type, create a locally provided and used type that is well-focused on your reference.
```

Comment:
It would be easier to follow structured like this:

New uses of this type are discouraged for several reasons:
* ignored fields...
* invalid usage help...
* inconsistent validation
* ...

Instead of using this type, create a local type that includes only the fields that you use.
For example, ... pointer to example we consider good ...